framework/test/jest/integration/utils/pg-helper/index.js

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright © 2019 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+
+const pgpLib = require('pg-promise');
+const childProcess = require('child_process');
+const { createStorageComponent } = require('./storage');
+
+this.pgpOptions = {
+	capSQL: true,
+	promiseLib: Promise,
+	noLocking: false,
+	noWarnings: true,
+};
+const pgp = pgpLib(this.pgpOptions);
+
+class PgHelper {
+	constructor({
+		dbName = 'lisk_dev',
+		user = 'lisk',
+		password = 'password',
+		host = 'localhost',
+		port = '5432',
+	}) {
+		this.dbName = dbName;
+		this.cnStr = `postgres://${user}:${password}@${host}:${port}/${dbName}`;
+		if (dbName.indexOf('postgres://') === 0) {
+			this.cnStr = dbName;
+		}
+
+		this.pgp = pgp(this.cnStr);
+		this.storage = null;
+	}
+
+	_dropDB() {
+		return new Promise(resolve => {
+			// eslint-disable-next-line no-console
+			console.log('Dropping database instance:', this.dbName);
+			childProcess.exec(`dropdb ${this.dbName}`, () => resolve());
+		});
+	}
+
+	_createDB() {
+		return new Promise((resolve, reject) => {
+			// eslint-disable-next-line no-console
+			console.log('Creating database instance:', this.dbName);
+			childProcess.exec(`createdb ${this.dbName}`, error => {
+				if (error) {
+					return reject(error);
+				}
+				return resolve();
+			});
+		});
+	}
+
+	async bootstrap() {
+		await this._dropDB();
+		await this._createDB();
+		this.conn = await this.pgp.connect();
+		return this.conn;
+	}
+
+	async cleanup() {
+		await this.storage.adapter.db.$pool.end();
+		await this.conn.done();
+		await this.pgp.$pool.end();
+	}
+
+	async createStorage(options = {}, logger) {
+		const storageOptions = {
+			database: this.dbName,
+			user: 'lisk',
+			password: 'password',
+			min: 1,
+			max: process.env.LISK_TEST_DB_MAX_CONNECTIONS || 2,
+			logFileName: `logs/devnet/lisk_${this.dbName}.log`,
+			noWarnings: true,
+			poolIdleTimeout: 30000,
+			reapIntervalMillis: 1000,
+			logEvents: ['error'],
+			...options,
+		};
+		this.storage = await createStorageComponent(storageOptions, logger);
+
+		return this.storage;
+	}
+
+	async createAccount(account) {
+		const keyValueSet = Object.keys(account).map(field => {
+			return {
+				field: `"${field}"`,
+				value:
+					field === 'publicKey'
+						? `DECODE(\${a.${field}}, 'hex')`
+						: `\${a.${field}}`,
+			};
+		});
+
+		await this.conn.query(
```

Comment:
Is this delete here to make sure there is no account with the same `publicKey`?

Reply:
Yes, publicKey is has a unique index. I didn't give too much thought on helper queries actually. I'd like to design them with the team. So, we can decide these on the STM that @sridharmeganathan will organize.